solver/src/interactions/uniswap.rs

```diff
@@ -36,42 +57,61 @@ mod tests {
     use super::*;
     use crate::encoding::tests::u8_as_32_bytes_be;
     use crate::interactions::dummy_web3;
+    use hex_literal::hex;
     use std::io::Cursor;
 
     #[test]
-    fn encode_uniswap_call_() {
+    fn encode_uniswap_call() {
         let amount_in = 5;
         let amount_out_min = 6;
-        let token_in = 7;
+        let token_in = H160::from_low_u64_be(7);
         let token_out = 8;
-        let payout_to = 9;
+        let payout_to = 9u8;
         let contract = UniswapV2Router02::at(&dummy_web3::dummy_web3(), H160::from_low_u64_be(4));
+        let settlement = GPv2Settlement::at(
+            &dummy_web3::dummy_web3(),
+            H160::from_low_u64_be(payout_to as u64),
+        );
         let interaction = UniswapInteraction {
             contract: contract.clone(),
+            settlement,
+            set_allowance: true,
             amount_in: amount_in.into(),
             amount_out_min: amount_out_min.into(),
-            token_in: H160::from_low_u64_be(token_in as u64),
+            token_in,
             token_out: H160::from_low_u64_be(token_out as u64),
-            payout_to: H160::from_low_u64_be(payout_to as u64),
         };
         let mut cursor = Cursor::new(Vec::new());
         interaction.encode(&mut cursor).unwrap();
-        let encoded = cursor.into_inner();
-        assert_eq!(&encoded[0..20], contract.address().as_fixed_bytes());
-        assert_eq!(encoded[20..23], [0, 1, 4]);
-        let call = &encoded[23..];
-        let signature = [0x38u8, 0xed, 0x17, 0x39];
+
+        // Verify Approve
+        let mut approve_call = cursor.into_inner();
+        assert_eq!(&approve_call[0..20], token_in.as_fixed_bytes());
+        assert_eq!(approve_call[20..23], [0, 0, 68]);
```

Comment:
Maybe a tiny comment that this is about the length of the calldata can help to understand it.

Reply:
I'd like to refactor these tests a bit to make them slightly easier to read and change later on. Will add a comment for now.